public/app/plugins/panel/timeseries/utils.ts

```diff
@@ -15,12 +15,14 @@ import { convertFieldType } from '@grafana/data/src/transformations/transformers
 import { GraphFieldConfig, LineInterpolation } from '@grafana/schema';
 import { applyNullInsertThreshold } from '@grafana/ui/src/components/GraphNG/nullInsertThreshold';
 import { nullToValue } from '@grafana/ui/src/components/GraphNG/nullToValue';
+import { findField } from 'app/features/dimensions';
 
 /**
  * Returns null if there are no graphable fields
  */
 export function prepareGraphableFields(
   series: DataFrame[],
+  xField: string | undefined,
```

Comment:
actually better... change this to a boolean flag (numericX=true?) and fail if series[0] is not number?